lms/djangoapps/instructor/tests/test_api.py

```diff
@@ -2443,6 +2443,16 @@ def assert_update_forum_role_membership(self, current_user, identifier, rolename
         elif action == 'revoke':
             assert rolename not in user_roles
 
+    def test_autoenroll_on_forum_role_add(self):
+        """
+        Test forum role modification auto enrolls user.
+        """
+        seed_permissions_roles(self.course.id)
+        user = UserFactory()
+        assert CourseEnrollment.is_enrolled(user, self.course.id) is False
+        self.assert_update_forum_role_membership(user, user.email, "Administrator", "allow")
```

Comment:
I don't think we did this in the previous PR, but could we do it here? 

```
                for rolename in ["Administrator", "Moderator", "Community TA"]:
                   self.assert_update_forum_role_membership(user, user.email, rolename, "allow")
                   assert CourseEnrollment.is_enrolled(user, self.course.id)
                   self.assert_update_forum_role_membership(user, user.email, rolename, "revoke")
```

You could also update the tests for the previous PR if you see fit.

Reply:
@mariajgrimaldi Done  :+1: